lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1401,9 +1401,25 @@ bool WitnessChecker::findBestWitness(
        attempt = static_cast<Attempt>(attempt + 1)) {
     SmallVector<ValueDecl *, 4> witnesses;
     switch (attempt) {
-    case Regular:
+    case Regular: {
       witnesses = lookupValueWitnesses(requirement, ignoringNames);
+
+      ValueDecl* thunk = nullptr;
+      for (auto witness : witnesses) {
+        if (auto func = dyn_cast<FuncDecl>(witness)) {
+          if (func->isDistributed()) {
+            thunk = func->getDistributedThunk();
+            fprintf(stderr, "[%s:%d] (%s) ALSO CONSIDER THUNK!!!\n", __FILE__, __LINE__, __FUNCTION__);
```

Comment:
So maybe this way we're able to consider and PICK the thunk as the witness?

Reply:
Just thinking if that would be a way to make calls "through the protocol" hit the thunk, by making the thunk the witness.

If this is the way we should do this we'd implement this somehow clean -- just trying to learn if this even makes sense.